vllm/logger.py

```diff
@@ -1,61 +1,255 @@
-# Adapted from
-# https://github.com/skypilot-org/skypilot/blob/86dc0f6283a335e4aa37b3c10716f90999f48ab6/sky/sky_logging.py
 """Logging configuration for vLLM."""
+import inspect
+import json
 import logging
 import os
 import sys
+import time
+import traceback
+from pathlib import Path
+
+from loguru import logger
 
 VLLM_CONFIGURE_LOGGING = int(os.getenv("VLLM_CONFIGURE_LOGGING", "1"))
 
-_FORMAT = "%(levelname)s %(asctime)s %(filename)s:%(lineno)d] %(message)s"
-_DATE_FORMAT = "%m-%d %H:%M:%S"
 
+class InterceptHandler(logging.Handler):
+    """A custom logging handler for intercepting logs and redirecting them.
+
+    This handler is designed to integrate with loguru's `logger` library,
+    capturing log records generated by Python's standard logging module and
+    processing them using `logger`.
+
+    Attributes:
+        loglevel_mapping (dict): Mapping of logging level numbers to string
+        representations.
+    """
+    loglevel_mapping = {
+        50: "CRITICAL",
+        40: "ERROR",
+        30: "WARNING",
+        20: "INFO",
+        10: "DEBUG",
+        0: "NOTSET",
+    }
+
+    def emit(self, record: logging.LogRecord):
+        """Process the logging record.
+
+        Overrides the default emit method to handle log records. It binds the
+        logger to a specific name and adds contextual information.
+
+        Args:
+            record (logging.LogRecord): The log record to process.
+        """
+        try:
+            level = logger.level(record.levelname).name
+        except AttributeError:
+            level = self.loglevel_mapping[record.levelno]
+
+        frame, depth = inspect.currentframe(), 0
+        while frame and (depth == 0
+                         or frame.f_code.co_filename == logging.__file__):
+            frame = frame.f_back
+            depth += 1
+
+        logger.opt(depth=depth,
+                   exception=record.exc_info).log(level, record.getMessage())
+
+
+class CustomizeLogger:
+    """A class for configuring and customizing logging.
+
+    Provides methods for setting up structured and unstructured logging,
+    including log rotation, retention, and formatting.
+
+    Methods are class methods, allowing direct access without instantiation.
+    """
+
+    @classmethod
+    def make_logger(cls, config_path: Path):
```

Comment:
It might be nice to add return annotations to the methods in this class as well:
```suggestion
def make_logger(cls, config_path: Path) -> loguru.Logger:
```

I see the contributing guidelines say 
>  Note: Currently, the repository does not pass the mypy tests.

and I think this would be required to make mypy happy, though I see a large portion of this repo does not have return type annotations :(